mesonbuild/interpreter.py

```diff
@@ -222,17 +222,29 @@ def __init__(self, subdir, sourcename, targetname, configuration_data):
 
 
 class EnvironmentVariablesHolder(MutableInterpreterObject, ObjectHolder):
-    def __init__(self, initial_values=None):
+    def __init__(self, initial_values=[], kwargs={}):
```

Comment:
__[Flake8]__ Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them. ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6229#issue-4887985))

<details>

| Rule |
| - |
| B006 |

</details>
<sub>posted by Sider</sub>